src/components/entity/ha-state-icon.js

```diff
@@ -14,11 +14,12 @@ class HaStateIcon extends PolymerElement {
       stateObj: {
         type: Object,
       },
+      overrideIcon: String
```

Comment:
All this component does is show an icon for a state, making it overridable is weird. If we know we want to override, we should not render this component but instead render a `ha-icon` directly.

Reply:
Are you ok with dropping `ha-state-icon` in `state-badge` and just always rendering `ha-icon` directly? `ha-state-icon` seems really redundant here anyway as all it's really doing is wrapping the `computeIcon()` call. Other components that need a state icon just use `state-badge` anyway

Reply:
Yeah, I'm fine with that.